internal/cli/cli.go

```diff
@@ -0,0 +1,71 @@
+package cli
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/kanopy-platform/buildah-plugin/internal/version"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+type RootCommand struct{}
+
+func NewRootCommand() *cobra.Command {
+	root := &RootCommand{}
+
+	cmd := &cobra.Command{
+		Use:               "buildah-plugin",
+		Short:             "Plugin for CI/CD tools to run buildah commands.",
+		PersistentPreRunE: root.persistentPreRunE,
+		RunE:              root.runE,
+	}
+
+	cmd.PersistentFlags().String("log-level", "info", "Configure log level")
+
+	cmd.AddCommand(newVersionCommand())
+
+	return cmd
+}
+
+func (c *RootCommand) persistentPreRunE(cmd *cobra.Command, args []string) error {
+	// additional settings based on plugin type
+	if err := pluginTypeSetup(); err != nil {
+		return err
+	}
+
+	// bind flags to viper
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+	viper.AutomaticEnv()
+
+	if err := viper.BindPFlags(cmd.Flags()); err != nil {
+		return err
+	}
+
+	// set log level
+	logLevel, err := log.ParseLevel(viper.GetString("log-level"))
+	if err != nil {
+		return err
+	}
+	log.SetLevel(logLevel)
+
+	return nil
+}
+
+func (c *RootCommand) runE(cmd *cobra.Command, args []string) error {
+	log.Infof("To be implemented... log-level: %v", viper.GetString("log-level"))
+	return nil
+}
+
+func pluginTypeSetup() error {
+	pluginType := version.Get().PluginType
+	switch pluginType {
+	case version.PluginTypeDrone:
```

Comment:
The plugin type is passed in via ldflags and drone specific set up is done here.
All drone plugin settings are passed in as environment variables with the prefix `PLUGIN_`.